### ISSUE TITLE
Suggestion: Prevent squishing of panes

### ISSUE THREAD

**Issue by ProLoser** [timomeh/hey-pane#15]
_As per @mrleblanc101's comments on #12 I'm creating a new issue for separate discussion._

I think it would be nice if you could set a 'minimum' dimension in absolute value that a pane will be reduced to. This way you can _try_ to resize a pane split to `90%` / `10%` but the smaller pane will not go below `200px`.

I personally set my separation to around `60%` so this isn't a big issue for me, but this seemed like a straightforward way to fix the problem for others.

> The real problem is that you squish the content making it look really unpleasant and distracting.
> There should be an absolute mode where the pane overlaps each other instead of being squished.
> It also causes conflict with the minimal plugin as you can see in the screenshot below.
> ![](https://user-images.githubusercontent.com/17392251/31369451-96d9d454-ad52-11e7-9f2a-d9a3969339c1.png)
> 
> Then setting pane looks good because it's content doesn't get squished or wrapped over multiple lines
> ![](https://user-images.githubusercontent.com/17392251/31369449-8dd04302-ad52-11e7-9eda-797dc89693d0.png)

**Comment by benyanke** [comment]
I would also find this helpful.

**Comment by ProLoser** [comment]
Also to address @mrleblanc101's original issue, I think his problem could be fixed by simply disabling wrapping.

**Comment by benyanke** [comment]
Good idea! Is that something that plugins can modify on the fly, pane-by-pane?

**Comment by timomeh** [comment]
_Because the issue is moved, I'll just copy my answer from the original Issue #12._

> That's not possible due to the way Panes work in Atom. The only way to not get squished content is by disabling soft wrap at the end of the Window and instead only wrap content at the 80char mark.

I checked the [Docs of Panes in Atom](https://atom.io/docs/api/v1.22.0/Pane) if it's possible to disable wrapping per pane, but I didn't find anything.

Also I saw in my config file that the setting for Text Wrapping is scoped to `editor` (it's `editor.softWrap` and `editor.softWrapAtPreferredLineLength`), which also makes me believe this is only a global editor setting and not adjustable per Pane.

Maybe someone has an idea how to achieve it nevertheless?

**Comment by benyanke** [comment]
Is it worth asking @atom to see if they'd be willing to implement pane-specific wrap overrides? That would seem to fit their goal of making the editor hackable.

**Comment by timomeh** [comment]
Opening a feature request is always a possibility, although I think supporting configs per pane is quite a big feature request. Also activating/deactivating soft wrap would only be a hack. For example, as you can see in the first screenshot of https://github.com/timomeh/hey-pane/issues/15#issue-271970920, the minimap is also squished, because the pane is so small.

We would need the possibility to let panes overlap, which also is quite a big feature request, because it would totally change the way panes are positioned.

**Comment by ProLoser** [comment]
I think people shouldn't use such extreme percentages to begin with. I personally care more about animated resizing than this story lol

**Comment by timomeh** [comment]
I personally use a "extreme" percentage, too, and I think many people also do.

**Comment by benyanke** [comment]
I would use an extreme percentage, but I can't because the wrapping doesn't work, and I want to be able to at least see a bit of the small column's file without it scrolling down super far when it compresses. If there were wrap disabling, I'd use an extreme percentages in a heartbeat.

Just my 2c and my usecase.
